File: app/src/main/java/com/robotnec/budget/core/mvp/presenter/TransactionsPresenter.java
package com.robotnec.budget.core.mvp.presenter;

import com.robotnec.budget.core.service.aggregation.AggregationService;
import com.robotnec.budget.core.service.aggregation.TransactionAggregation;
import com.robotnec.budget.core.persistence.dao.TransactionDao;
import com.robotnec.budget.core.di.ApplicationComponent;
import com.robotnec.budget.core.domain.operation.Transaction;
import com.robotnec.budget.core.mvp.view.TransactionsView;
import com.robotnec.budget.core.navigator.NavigationBundle;
import com.robotnec.budget.core.navigator.Navigator;

import java.util.List;

import javax.inject.Inject;

/**
 * @author zak <zak@swingpulse.com>
 */
public class TransactionsPresenter extends Presenter<TransactionsView> {

    @Inject
    Navigator navigator;

    @Inject
    TransactionDao transactionDao;

    @Inject
    AggregationService aggregator;

    public TransactionsPresenter(TransactionsView view) {
        super(view);
    }

    @Override
    public void onViewReady() {
        List<Transaction> transactions = transactionDao.getAll();
        TransactionAggregation aggregation =
                aggregator.aggregate(transactions, AggregationService.Resolution.DAY);
        view.displayTransactions(aggregation);
    }

    @Override
    public void injectComponent(ApplicationComponent applicationComponent) {
        applicationComponent.inject(this);
    }

    public void addNewTransaction(NavigationBundle navigationBundle) {
        navigator.openAddTransactionScreen(navigationBundle);
    }
}

File: app/src/main/java/com/robotnec/budget/core/mvp/presenter/TransactionsPresenter.kt
package com.robotnec.budget.core.mvp.presenter

import com.robotnec.budget.refactored.di.ApplicationComponent
import com.robotnec.budget.core.mvp.view.TransactionsView
import com.robotnec.budget.core.navigator.NavigationBundle
import com.robotnec.budget.core.navigator.Navigator
import com.robotnec.budget.core.persistence.dao.TransactionDao
import com.robotnec.budget.core.service.aggregation.AggregationService
import javax.inject.Inject

/**
 * @author zak zak@swingpulse.com>
 */
class TransactionsPresenter(view: TransactionsView) : Presenter<TransactionsView>(view) {

    @Inject
    internal lateinit var navigator: Navigator

    @Inject
    internal lateinit var transactionDao: TransactionDao

    @Inject
    internal lateinit var aggregator: AggregationService

    override fun onViewResume() {
        val transactions = transactionDao.all
        val aggregation = aggregator.aggregate(transactions, AggregationService.Resolution.DAY)
        view.displayTransactions(aggregation)
    }

    override fun injectComponent(applicationComponent: ApplicationComponent) {
        applicationComponent.inject(this)
    }

    fun addNewTransaction(navigationBundle: NavigationBundle<*>) {
//        navigator.openAddTransactionScreen()
    }
}
